weaviate/collections/collection/async_.py

```diff
@@ -103,9 +105,12 @@ def __init__(
 
     async def length(self) -> int:
         """Get the total number of objects in the collection."""
-        total = (await self.aggregate.over_all(total_count=True)).total_count
-        assert total is not None
-        return total
+        if self._connection._weaviate_version.is_lower_than(1, 28, 4):
```

Comment:
do we need this? Shouldn't `self.aggregate.over_all ` be over GRPC when it is available and over GQL when it is not?

Reply:
Yes good point. This was the implementation before the rest of the changes were made, forgot to change it back!